Rework modules

chrisr3d opened the pull request:
Rework of the expansion and import module system, allowing them to return MISP attributes & objects as result.

- Including a new view to visualize the module results, using the same style as the event view
- Current system returning freetext not impacted and still running
- This code will not be raised until some [new format modules](https://github.com/MISP/misp-modules/tree/new_module) are merged in the misp-modules master branch
- New objects & attributes (from the module results) added into the event do not overwrite the already existing ones (even when the values are the same)

Some love for a full review is needed

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment] Testing if an object has attributes before looping over them

chrisr3d committed:
fix: [enrichment] Avoiding errors on distribution and sharing group data

- Making sure we do have a value for distribution
  and sharing group
- Avoid translating distribution levels into their
  human meaning because the output process capturing
  data from the form will anyway use the numeric
  values. Readable values are btw displayed in the
  form for users confort
- Also sanitized data displayed

chrisr3d committed:
fix: [enrichment view] Trying to avoid submit button to be somewhere not intended

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment view] Fixed Object sharing group field location

chrisr3d committed:
fix: [enrichment view] Added missing form fields + button typo

- Those form inputs are the ones that are used in
  the javascript part to submit results

chrisr3d committed:
fix: [enrichment] Set sharing group id to 0 when distribution is not 'Sharing Group'

- Also setting some return data & fields

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment] Fixed missing merge conflict

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment] Fixed handleModuleResults input variable name

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [cleanup] Fixed indentations

chrisr3d committed:
fix: [freetext import] Deleted not used variable

chrisr3d committed:
wip: [enrichment] Capturing objects & attributes

chrisr3d committed:
fix: [enrichment view] Quick display fix

- Making sure the submit button is always at the
  bottom of the page
- Fixed missing closing tags

chrisr3d committed:
fix: [enrichment] Fixed missing object_relation field

... for object attributes fetched from module
results form

chrisr3d committed:
add: [enrichment] Added possibility to get object template version & uuid

chrisr3d committed:
wip: [enrichment] Saving attributes & objects from module results

- Need to handle specific cases, relationships,
  and to update the progress status information

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
chg: [enrichment] Defined function to appy inflector on strings

So we can use it each time we want to display the
freetext import / module results status message

chrisr3d committed:
wip: [enrichment] Returning a status message after the module results handling is done

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment] Typo on Object references field relationship_type

chrisr3d committed:
chg: [ObjectReference] Referenced id, uuid & type set now from a function that could be reused

chrisr3d committed:
wip: [enrichment] Support of object references

- Handling the references between objects and
  attributes or objects that are displayed in the
  form and saving them.
- Avoiding issue when an attribute or object can
  not be saved and is referenced: trying to find
  in the event if it is an attribute, returning
  a warning message if it is an object or if
  nothing corresponding to the failing referenced
  object or attribute is found.

chrisr3d committed:
fix: Removed test variables

- Also pardon my French

chrisr3d committed:
chg: [enrichment] Passing the initial object to the form

- If the attribute we query the module on is an
  object attribute, we want to have this object
  information se we do not add or overwrite
  attributes already here

chrisr3d committed:
fix: [enrichment] Displaying error value in the job status instead of the json format

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment view] Quick change, using variable

... Instead of going twice through the dictionary
keys to fetch the same value

chrisr3d committed:
wip: [hover enrichment] Passing new modules results to the hover enrichment view

chrisr3d committed:
add: [hover enrichment] Displaying resolved objects & attributes from new modules results

- Also quick indentation fix

chrisr3d committed:
fix: [enrichment] Simplified tests if not empty

- In those specific cases, we can simply skip
  testing if something is set or equals 0, '' or
  false, since it is all implied in 'empty'

chrisr3d committed:
wip: [enrichment] Passing the initial object within the request data

- Makes its parsing easier afterwards

chrisr3d committed:
wip: [enrichment] Passing initial object references as well

- Also testing if the initial object found is not empty

chrisr3d committed:
wip: [enrichment] Avoiding duplicate object attributes

- It concerns obviously the case where we query a
  module using an attribute within an object as input
- More to come about the ObjectReference field that
  should not be overwritten/duplicated either

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
wip: [enrichment] Handling the potential uuids differences

- We want to be sure the references we add to an
  event are pointing to the right target, so
  when an attribute/object is already in the event
  and is a reference target, we want to point to
  this already existing attribute/object, and not
  to the one we get from the module results, which
  will afterall be skipped.
- Also to to that, attributes already in the event
  are simply not saved, but we need to find in the
  event if an event already exists or not, using
  its attributes
- More care to the references themselves to come

chrisr3d committed:
fix: [enrichment] Fixed inflector typo

chrisr3d committed:
fix: [enrichment] Redefinition of the '$failed' array

- Used as list and not as dict
- Used to keep a list of failing attribute & objects
- Attributes already present in the event are no
  longer saved in this list, and their uuid is saved
  in the '$recovered_uuids' list which is used to
  redirect the resolved attribute / object uuid to
  the same already present attribute / object

chrisr3d committed:
fix: [enrichment] Better resolved objects handling

- We handle the initial object independently from
  all the other objects.
- We make sure Object attributes are not already
  in the event (i.e the object is not already in
  the event) before saving it
- Some other cases handled a better way to make
  sure we skip saving object attributes or
  references when they should not be, or to skip
  what should be skipped.

chrisr3d committed:
fix: [enrichment] Better references handling

- References are all handled at the end, after the
  attributes and objects are all saved (or skipped
  when needed).
- Since we no longer have any data in '$failed',
  we simply skip every reference related to uuids
  present in this list.
- '$recovered_uuids' is the list used to handle
  uuids of resolved attributes/objects already
  in the event.
- We also skip references already present in the
  event (i.e same source, same target and same
  relationship type).

chrisr3d committed:
fix: [enrichment] Displaying reference saving errors in the job status

- Finaly using the dedicated list to display
  validation errors for references

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [hover enrichment] Displaying information of empty results when needed

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment] Removed possibility to display freetext result in the new module results view

chrisr3d committed:
chg: [enrichment] Making attributes category & type changeable

- To keep possibility to select the attribute type
  or category, as it is in freetext import
- Category and Type values that can be selected by
  users should be specified in the results, using
  an array. Otherwise, they are set and cannot be
  changed in the view (as it is the case for the
  other fields like value and uuid)
- This feature is only applicable on attributes
  level, and not on object attributes level

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment] Removing traces of some variables related to the freetext import

chrisr3d committed:
fix: [enrichment] Fixed change on types & categories

... that should be applied on attributes level and
not on object attributes level

chrisr3d committed:
fix: [enrichment view] Using some MISP event view parts

- Using the same design as objects and attributes as in
  the MISP event view

chrisr3d committed:
fix: [enrichment] Fixed tags classes names following the latest changes on the module results view

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment view] Fixed mispositioned closing tags

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment] Json format of validation errors

chrisr3d committed:
fix: [enrichment] Setting importComment value before going to the view

- Saving a variable & will avoid issues when reusing
  the view for import modules

chrisr3d committed:
chg: [enrichment] Import modules now support new modules format

- Import module results are sent to the new view
  (same as for expansion modules) whenever they
  support objects import
- Function handling the results right after they
  are returned from the module has also been moved
  to Model/Event and is used for both expansion &
  import modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment view] Displaying forgotten header when there is no object in module results

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment view] Fixed attributes display wwwwwwwwwww

- Making attribute value display accept to be on
  multilines to let more space for uuids to be on
  1 line
- This makes the global display lighter with long
  attribute values, because they were displayed on
  only 1 line, making uuids displayed on 3 or 4
  lines, making the complete view height way bigger
  than expected

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: Returning tags & galaxies from module results

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
add: [enrichment view] Displaying events metadata & number of returned attributes

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
add: [enrichment view] Displaying event level tags

chrisr3d committed:
add: [enrichment] Attaching event level tags returned by the module results

chrisr3d committed:
add: [enrichment] Displaying + returning tags at attributes level from the module results

- If one of our UI master could review this and find
  a better way of displaying the tags at attribute
  level in the module results view, it would be
  very cool :D

chrisr3d committed:
fix: [enrichment] Added missing variable

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment view] Displaying Tags only if not empty (error otherwise)

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

adulau commented:
FYI, [misp-modules](https://github.com/MISP/misp-modules) is updated with the new sample modules using the new format and the previous modules operate as expected. I'm ready to start tagging misp-modules (as 2.4.110) when this branch is merged in MISP core.

chrisr3d committed:
add: [enrichment view] Added data fields + cancel button to the form

chrisr3d committed:
add: [enrichment] Fetching data fields from the view

chrisr3d committed:
fix: [enrichment] Strengthened query conditions to find existing object attributes

chrisr3d committed:
fix: [enrichment] Applying regular expressions to values before displaying them

- So when the DB is queried to find if object
  attributes with those values already exist,
  we do not miss some stored values

chrisr3d committed:
chg: Bumped query version

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
Merge branch '2.4' of https://github.com/MISP/MISP into rework_modules

chrisr3d committed:
fix: [enrichment view] Testing if event metadata elements exist before displaying them
